compiler/src/iree/compiler/Codegen/LLVMGPU/ConvertToLLVM.cpp

```diff
@@ -309,6 +309,19 @@ class ConvertIREEBindingSubspanOp : public ConvertToLLVMPattern {
     llvmFuncOp.setArgAttr(llvmBufferArg.getArgNumber(),
                           LLVM::LLVMDialect::getAlignAttrName(),
                           rewriter.getI32IntegerAttr(16));
+    // The ranges used within the bindings are guaranteed not to alias
+    llvmFuncOp.setArgAttr(llvmBufferArg.getArgNumber(),
+                          LLVM::LLVMDialect::getNoAliasAttrName(),
+                          rewriter.getUnitAttr());
+    if (subspanOp.getDescriptorFlags().has_value() &&
+        ((subspanOp.getDescriptorFlags().value() &
```

Comment:
you can use `bitEnumContainsAny` here as
```c++
if (bitEnumContainsAny(
      subspanOp.getDescriptorFlags().value_or(IREE::HAL::DescriptorFlags::None),
      IREE::HAL::DescriptorFlags::ReadOnly)) {
```
(there's also a bitEnumContainsAll that is useful as well)

Reply:
Thanks, this is much better. My code was looking ugly.